src/main/java/org/fcrepo/migration/handlers/BasicObjectVersionHandler.java

```diff
@@ -175,12 +162,210 @@ public void processObjectVersions(Iterable<ObjectVersionReference> versions) {
     }
 
     private boolean isDateProperty(String uri) {
-        return uri.equals("info:fedora/fedora-system:def/model#createdDate") || uri.equals("info:fedora/fedora-system:def/view#lastModifiedDate");
-
+        return uri.equals("info:fedora/fedora-system:def/model#createdDate") ||
+               uri.equals("info:fedora/fedora-system:def/view#lastModifiedDate") ||
+               uri.equals("http://www.loc.gov/premis/rdf/v1#hasDateCreatedByApplication") ||
+               uri.equals("http://www.loc.gov/premis/rdf/v1#hasEventDateTime");
     }
 
     private FedoraObject createObject(ObjectReference object) throws FedoraException {
         return repo.createObject(idMapper.mapObjectPath(object));
     }
 
+    private void updateObjectProperties(ObjectVersionReference version,
+                                        FedoraObject object,
+                                        QuadAcc triplesToRemove,
+                                        QuadDataAcc triplesToInsert) {
+        if (version.isLastVersion()) {
+            for (ObjectProperty p : version.getObjectProperties().listProperties()) {
+                String pred = p.getName();
+                String obj = p.getValue();
+
+                // Map dates and object state
+                if (pred.equals("info:fedora/fedora-system:def/model#createdDate")) {
+                    pred = "http://www.loc.gov/premis/rdf/v1#hasDateCreatedByApplication";
+                } else if (pred.equals("info:fedora/fedora-system:def/model#state")) {
+                    pred = "http://fedora.info/definitions/1/0/access/objState";
+                } else if (pred.equals("info:fedora/fedora-system:def/view#lastModifiedDate")) {
+                    pred = "http://www.loc.gov/premis/rdf/v1#hasEventDateTime";
+
+                    // Add the migration event type for the last modified date
+                    updateTriple(triplesToRemove,
+                                 triplesToInsert,
+                                 "http://www.loc.gov/premis/rdf/v1#hasEventType",
+                                 "migration");
```

Comment:
You post RDF with a blank node in it, e.g.:

<> dcterms:creator [
    foaf:name "Bob Smith"
]